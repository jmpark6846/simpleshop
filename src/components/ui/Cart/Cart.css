.cart{ 
  position: absolute;
  background-color: white;
  top: 53px;
  right: 15px;
  border:1px solid lightgray;
  border-radius: 3px;
  display: none;
  font-size: 0.9rem;
  width: 300px;
  z-index: 1000;
}


.cart.show{
  display: block;
}

.cart-item, .cart-item-wrapper{
  padding: 15px;
  
  border-bottom: 1px solid lightgray;
  display: flex;
}
.cart-item .image{
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width:30%;
  display: inline-block;
}

.cart-item .image::after{
  /* .slider .slide::after{ */
    content:'';
    display: block;
    z-index: -100;
    padding-bottom: 100%;
  }

.cart-item .item-info{
  width:70%;
  display: inline-block;
  font-size: 0.9rem;
  vertical-align: top;
  padding: 0 5px;
}

.cart .error{
  background-color: rgba(216, 112, 147);
  color:white;
}

.cart .icon{
  margin-right: 5px;
}
.cart .delete{
  align-self: center;
}
.cart .delete .icon{
  cursor: pointer;
}

.cart button.buy{
  width:100%;
}