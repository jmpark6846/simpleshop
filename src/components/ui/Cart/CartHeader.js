import React from 'react'
import PropTypes from 'prop-types'

const CartHeader = ({value}) => {
  return (
    <div className='cart-header'>
      {value}
    </div>
  )
}

CartHeader.propTypes = {
  value: PropTypes.string
}

export default CartHeader
