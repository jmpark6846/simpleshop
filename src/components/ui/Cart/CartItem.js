import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { NavLink } from "react-router-dom";
import NumberSelector from '../../ui-components/NumberSelector/NumberSelector'
import Price from "../../ui-components/Price/Price";
import Icon from '../../ui-components/Icon/Icon'
import { doDeleteCartItem, doChangeEA } from '../../../actions/cart';

export const CartItem = ({cartItem, deleteCartItem, changeEA}) => {
  return (
    <div className='cart-item'>
      <div className='image' style={{ backgroundImage: `url(${cartItem.img})` }}></div>      
      <div className='item-info'>
        <div><NavLink to={'/product/'+cartItem.id} >{ cartItem.name.length > 20 ? `${cartItem.name.substr(0,20)}...` : cartItem.name }</NavLink></div>
        <div><Price price={ cartItem.price } suffix='원' /></div>
        <NumberSelector number={cartItem.ea} onChange={(ea)=>changeEA(cartItem.id, ea)} />
      </div>
      <div className='delete'>
        <Icon icon='far fa-trash-alt' onClick={()=>deleteCartItem(cartItem.id)} />
      </div>
    </div>
  )
}

CartItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  changeEA: PropTypes.func,
  deleteCartItem: PropTypes.func,
  cartItem: PropTypes.object
}

const mapDispatch = (dispatch) => ({
  changeEA: (id, ea) => dispatch(doChangeEA(id, ea)),
  deleteCartItem: (id) => dispatch(doDeleteCartItem(id))
})
export default connect(undefined, mapDispatch)(CartItem)
