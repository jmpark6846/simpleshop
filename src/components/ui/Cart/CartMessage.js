import React from 'react'
import PropTypes from 'prop-types'
import Icon from '../../ui-components/Icon/Icon';

class CartMessage extends React.Component{
  render(){
    const { children, show, handleCloseClick } = this.props
    return (
      <div className={ show ? 'cart-message' : 'cart-message hide'} >
        <div className='cart-message-wrapper'>
          <div className='cart-message-body'>
            { children ? children : '장바구니에 추가되었습니다.'}
            <span onClick={handleCloseClick} className='icon'>×</span>
          </div>
          <i className='chat-start-icon'></i>
        </div>
      </div>
    )
  }
  
}

CartMessage.propTypes = {
  children: PropTypes.node,
  show: PropTypes.bool,
  handleCloseClick: PropTypes.func
}

export default CartMessage
