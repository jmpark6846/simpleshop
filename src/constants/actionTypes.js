// cart
export const ADD_TO_CART = 'ADD_TO_CART'
export const DELETE_CART_ITEM = 'DELETE_CART_ITEM'
export const EA_CHANGE = 'EA_CHANGE'


// product
export const PRODUCT_LOAD = 'PRODUCT_LOAD'
export const REVIEW_ADD = 'REVIEW_ADD'
export const REVIEW_DELETE = 'REVIEW_DELETE'


// order
export const ORDER_ADD = 'ORDER_ADD'
export const ORDER_DELETE = 'ORDER_DELETE'
export const ORDER_PROCESS = 'ORDER_PROCESS'


// common
export const LOAD_NEW_PAGE = 'LOAD_NEW_PAGE'

