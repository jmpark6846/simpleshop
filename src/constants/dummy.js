export const products = [
  { 
    id:0, 
    name: 'Adventurist Daypack - Black', 
    price: 590000,
    thumbnail: '/src/asset/img/1.jpeg', 
    imgs: ['/src/asset/img/1.jpeg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], 
    rating: 3.66, 
    reviewCount: 3,
    reviews: [ 
      { id: 0, title: '타이틀', content: '리뷰내용이다', rating: 3 },
      { id: 1, title: '타이틀', content: '리뷰내용이다', rating: 4 },
      { id: 2, title: '타이틀', content: '리뷰내용이다', rating: 4 },
    ] 
  },
  // { id:1, name: 'Adventurist Daypack - Pine', price: 5800000, imgs: ['/src/asset/img/2.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },   
  // { id:2, name: 'Adventurist Daypack - Sand', price: 60000, imgs: ['/src/asset/img/3.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },
  // { id:3, name: 'Adventurist Daypack - Black', price: 610, imgs: ['/src/asset/img/1.jpeg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },  
  // { id:4, name: 'Adventurist Daypack - Pine', price: 6200, imgs: ['/src/asset/img/2.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },   
  // { id:5, name: 'Adventurist Daypack - Sand', price: 6300000, imgs: ['/src/asset/img/3.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },
  // { id:6, name: 'Adventurist Daypack - Black', price: 6400, imgs: ['/src/asset/img/1.jpeg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },  
  // { id:7, name: 'Adventurist Daypack - Pine', price: 65000, imgs: ['/src/asset/img/2.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 },   
  // { id:8, name: 'Adventurist Daypack - Sand', price: 66000, imgs: ['/src/asset/img/3.jpg', '/src/asset/img/2.jpg', '/src/asset/img/3.jpg'], rating: 3.75, reviewCount: 4 }
]

