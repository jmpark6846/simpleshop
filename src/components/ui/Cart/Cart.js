import React from 'react'
import PropTypes from 'prop-types'
import { connect } from "react-redux";
import classNames from 'classnames'
import './Cart.css'
import CartItem from './CartItem';

import Icon from '../../ui-components/Icon/Icon';
import Button from '../../ui-components/Button/Button';
import Price from '../../ui-components/Price/Price';

export const Cart = ({show, cartItems, totalPrice}) => {
  return (
    <div className='cart-wrapper'>
      <Button className='shopping-cart' icon='fas fa-shopping-cart'/>
      <div className='cart-preview-div'>
        <div className={classNames('cart', { show })} >
          { cartItems &&Object.keys(cartItems).length !== 0 ? 
            Object.keys(cartItems).map(id=> <CartItem key={id} cartItem={cartItems[id]} />)          
            :
            <div className='cart-item-wrapper cart-empty-message'>장바구니가 비었습니다.</div> 
          }
        
          <div className='cart-item-wrapper'>
            총 금액: <Price className='cart-total-price' price={totalPrice} suffix='원'/>
          </div> 
          <div className='cart-item-wrapper'>
            <Button className='buy' value='구매하기' primary/>
          </div>
        </div>
      </div>  
    </div>
  )
}

Cart.propTypes = {
  show: PropTypes.bool,
  cartItems: PropTypes.object,
  totalPrice: PropTypes.number
}

const mapState = ({cart}) => ({
  cartItems: cart.cartItems,
  totalPrice: cart.totalPrice
})

export default connect(mapState)(Cart)
