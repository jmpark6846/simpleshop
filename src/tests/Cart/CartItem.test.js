import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store';

import { CartItem } from '../../components/ui/Cart/CartItem';
import ConnectedCartItem from '../../components/ui/Cart/CartItem';

import { products } from "../../constants/dummy";

Enzyme.configure({ adapter: new Adapter()})

describe('CartItem', () => {
  const mockStore = configureMockStore();
  const cartItem = { 
    id:0, 
    name: 'bag', 
    price: 100,
    img: '/path/to/bag.jpg', 
  }

  let store = mockStore({
    cart: {
      cartItems: { 0 : cartItem },
      totalPrice: cartItem.price
    }
  })


  it('CartItem snapshot test', () => {
    const tree = renderer.create(<CartItem cartItem={cartItem} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('CarItem rendering test', () => {
    const wrapper = shallow(<CartItem cartItem={cartItem} />)
    expect(wrapper.find('.image').props().style).toEqual({ backgroundImage: `url(${cartItem.img})` })
  })
})
