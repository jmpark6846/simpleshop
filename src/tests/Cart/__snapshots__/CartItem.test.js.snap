// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem CartItem snapshot test 1`] = `
<div
  className="cart-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(/path/to/bag.jpg)",
      }
    }
  />
  <div
    className="item-info"
  >
    <div>
      bag
    </div>
    <div>
      <span>
        100
        Ïõê
      </span>
    </div>
    <div
      className="number-selector"
    >
      <button
        className="decrement"
        onClick={[Function]}
      >
        -
      </button>
      <input
        className="number"
        name="number"
        onChange={[Function]}
      />
      <button
        className="increment"
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
  <div
    className="delete"
  >
    <span
      className="icon far fa-trash-alt"
      onClick={[Function]}
    />
  </div>
</div>
`;
